lib/docker/image/image_name_test.go

```diff
@@ -54,6 +54,30 @@ func TestParseImageName(t *testing.T) {
 	require.True(name.IsValid())
 	require.Equal("127.0.0.1:5002/evanescence-golang-1:latest", name.String())
 
+	name, err = ParseNameForPull("127.0.0.1:5002/king-gizzard-golang-1@sha256:2a3dd484ecfcf9343994e0f6c2af0a6faf1af7f7e499905793643f91e90edcb3")
+	require.NoError(err)
+	require.Equal(name.GetRegistry(), "127.0.0.1:5002")
+	require.Equal(name.GetRepository(), "king-gizzard-golang-1")
+	require.Equal(name.GetTag(), "sha256:2a3dd484ecfcf9343994e0f6c2af0a6faf1af7f7e499905793643f91e90edcb3")
+	require.True(name.IsValid())
+	require.Equal("127.0.0.1:5002/king-gizzard-golang-1@sha256:2a3dd484ecfcf9343994e0f6c2af0a6faf1af7f7e499905793643f91e90edcb3", name.String())
+
+	name, err = ParseNameForPull("127.0.0.1:5002/king-gizzard-golang-1:v1.0.0@sha256:2a3dd484ecfcf9343994e0f6c2af0a6faf1af7f7e499905793643f91e90edcb3")
+	require.NoError(err)
+	require.Equal(name.GetRegistry(), "127.0.0.1:5002")
+	require.Equal(name.GetRepository(), "king-gizzard-golang-1")
+	require.Equal(name.GetTag(), "sha256:2a3dd484ecfcf9343994e0f6c2af0a6faf1af7f7e499905793643f91e90edcb3")
+	require.True(name.IsValid())
+	require.Equal("127.0.0.1:5002/king-gizzard-golang-1@sha256:2a3dd484ecfcf9343994e0f6c2af0a6faf1af7f7e499905793643f91e90edcb3", name.String())
```

Comment:
note how the informational tag was dropped here -- i see this behaviour as fine, as i think it's basically syntactic sugar, but let me know if you'd prefer it to be kept and i can refactor

Reply:
I am okay with this behavior.